facebook/micro/monthly-archive.sh

```diff
@@ -8,13 +8,20 @@ else
 fi
 echo ${MONTH}
 R_MONTH=${MONTH#*_}; R_MONTH=${R_MONTH#0}
-BATCH="cd fb-public-results\nls -1 cvid_responses_${MONTH}*.gz"
-sftp -b <(echo -e "${BATCH}") -P 2222 fb-automation@ftp.delphi.cmu.edu 2>/dev/null | \
-    grep "^cvid" | \
-    awk -F_ 'BEGIN{print "cd fb-public-results"} {key=$3 $4 $5; if (key!=last && last!="") {print record} last=key; record=$0} END{print record}' | \
-    sed '/^cvid/ s/^/get /' >fetch.sftp
-sftp -b fetch.sftp -P 2222 fb-automation@ftp.delphi.cmu.edu
-OUT=${MONTH/_/-}
-Rscript ../monthly-files.R ${MONTH%_*} ${R_MONTH} . >${OUT}.csv
-gzip ${OUT}.csv
-sftp -b <(echo -e "cd fb-public-results\nput ${OUT}.csv.gz") -P 2222 fb-automation@ftp.delphi.cmu.edu
+
+perform_rollup_and_post ()
+{
+    BATCH="cd $1\nls -1 cvid_responses_${MONTH}*.gz"
+    sftp -b <(echo -e "${BATCH}") -P 2222 fb-automation@ftp.delphi.cmu.edu 2>/dev/null | \
+        grep "^cvid" | \
+        awk -F_ 'BEGIN{print "cd '"$1"'"} {key=$3 $4 $5; if (key!=last && last!="") {print record} last=key; record=$0} END{print record}' | \
```

Comment:
On the topic of single quotes changing variable behavior, I want to highlight this change:

```
awk -F_ 'BEGIN{print "cd '"$1"'"}...'
```

was previously

```
awk -F_ 'BEGIN{print "cd fb-public-results"}...'
```

, which requires no variable expansion. To allow expansion of positional argument 1, I placed it it its own double-quoted string chunk to get concatenated with the rest of the `awk` processing command. A small test example seems to work correctly:

```
myvar="fb-public-results"
awk -F_ 'BEGIN{print "cd '"$myvar"'"}'
```

Reply:
Thanks for raising. The canonical way to get environment variables into an awk program is this:
```suggestion
        awk -F_ -vDIR="$1"  'BEGIN{print "cd " DIR} {key=$3 $4 $5; if (key!=last && last!="") {print record} last=key; record=$0} END{print record}' | \
```

You can test with
```
thing="foo"; awk -vDIR="$thing" 'BEGIN{print "cd " DIR}'
# cd foo
```